packages/shared/src/hooks/post/useSmartTitle.ts

```diff
@@ -87,13 +96,34 @@ export const useSmartTitle = (post: Post): UseSmartTitle => {
     ...disabledRefetch,
   });
 
+  const { fetchTranslations } = useTranslation({
+    clickbaitShieldEnabled: !clickbaitShieldEnabled,
+  });
+
   const fetchSmartTitle = useCallback(async () => {
     if (!fetchedSmartTitle) {
-      await refetch();
-    } else {
-      client.setQueryData(key, post?.title);
+      const smartTitlePost = {
+        ...post,
+        ...smartTitle,
+      };
+
+      const [translateResult] = await fetchTranslations([smartTitlePost]);
+
+      if (translateResult) {
+        client.setQueryData(
+          key,
+          updateTitleTranslation({
+            post: smartTitlePost,
+            translation: translateResult,
+          }),
+        );
+      } else {
+        await refetch();
+      }
```

Comment:
if translation was returned, update cache, if not ask API for fresh copy (which is already translated)

Reply:
you mean to not refetch each time?

Reply:
`refetch` only calls API once, then it just re-uses its cache (`staleTime: Infinity`)